Fatras/include/ActsFatras/Physics/PhotonConversion/PhotonConversion.hpp

```diff
@@ -0,0 +1,311 @@
+/// after which the interaction will occur and the final state due the
+/// interaction itself.
+class PhotonConversion {
+ public:
+  using Scalar = ActsFatras::Particle::Scalar;
+  /// Lower energy threshold for produced children
+  Scalar minimumMomentum = 100. * Acts::UnitConstants::MeV;
+  /// Scaling factor of children energy
+  Scalar childEnergyScaleFactor = 2.;
+  /// Scaling factor for photon conversion probability
+  Scalar conversionProbScaleFactor = 0.98;
+
+  /// Method for evaluating the distance after which the photon
+  /// conversion will occur.
+  ///
+  /// @tparam generator_t Type of the random number generator
+  /// @param [in, out] generator The random number generator
+  /// @param [in] momentum The momentum of the particle
+  ///
+  /// @return valid X0 limit and no limit on L0
+  template <typename generator_t>
+  std::pair<Scalar, Scalar> generatePathLimits(generator_t& generator,
+                                               const Particle& particle) const;
+
+  /// This method evaluates the final state due to the photon conversion.
+  ///
+  /// @tparam generator_t Type of the random number generator
+  /// @param [in, out] generator The random number generator
+  /// @param [in, out] particle The interacting photon
+  /// @param [out] generated List of generated particles
+  ///
+  /// @return True if the conversion occured, else false
+  template <typename generator_t>
+  bool run(generator_t& generator, Particle& particle,
+           std::vector<Particle>& generated) const;
+
+ private:
+  /// This method constructs and returns the child particles.
+  ///
+  /// @param [in] photon The interacting photon
+  /// @param [in] childEnergy The energy of one child particle
+  /// @param [in] childDirection The direction of the child particle
+  ///
+  /// @return Array containing the produced leptons
+  std::array<Particle, 2> generateChildren(
+      const Particle& photon, Scalar childEnergy,
+      const Particle::Vector3& childDirection) const;
+
+  /// Generate the energy fraction of the first child particle.
+  ///
+  /// @tparam generator_t Type of the random number generator
+  /// @param [in, out] generator The random number generator
+  /// @param [in] gammaMom The momentum of the photon
+  ///
+  /// @return The energy of the child particle
+  template <typename generator_t>
+  Scalar generateFirstChildEnergyFraction(generator_t& generator,
+                                          Scalar gammaMom) const;
+
+  /// Generate the direction of the first child particle.
+  ///
+  /// @tparam generator_t Type of the random number generator
+  /// @param [in, out] generator The random number generator
+  /// @param [in] particle The photon
+  ///
+  /// @return The direction vector of the child particle
+  template <typename generator_t>
+  Particle::Vector3 childDirection(generator_t& generator,
+                                   const Particle& particle) const;
```

Comment:
```suggestion
  Particle::Vector3 generateChildDirection(generator_t& generator,
                                   const Particle& particle) const;
```
Consistent naming.

Reply:
Changed to `generateFirstChildDirection`